plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -626,6 +650,9 @@ public LocatedTableHandle getTableHandle(
             return null;
         }
         verifySupportedColumnMapping(getColumnMappingMode(metadataEntry, protocolEntry));
+        if (cacheTableMetadata && endVersion.isEmpty() && !isCachedVersionSameAsLastTransaction(metastoreTable.get(), tableSnapshot)) {
```

Comment:
In case of referencing multiple times the same table in a query, is this check supposed to avoid calling metastore table replace repeatedly for the same table?

I don't follow where we ensure that the call is made only once.

I'm thinking that this should be ensured with the help of a ConcurrentMap like in `io.trino.plugin.deltalake.DeltaLakeMetadata#getSnapshot(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTableName, java.lang.String, java.util.Optional<java.lang.Long>)`